proposals/0255-Enhance-BodyInformation-vehicle-data.md

```diff
@@ -16,118 +16,161 @@ In order to partner with more diverse app partners, we need to provide additiona
 ## Proposed Solution 
 
 Following vehicle data params are added to `BodyInformation` struct:
-* `trunkAjar`
-* `hoodAjar`
-* `frontLeftDoorLocked`
-* `frontRightDoorLocked`
-* `rearLeftDoorLocked`
-* `rearRightDoorLocked`
-* `trunkLocked`
+* `doorsStatus`
+* `gatesStatus`
+
+Following vehicle data params are deprecated from `BodyInformation` struct:
+* `driverDoorAjar`
+* `passengerDoorAjar`
+* `rearLeftDoorAjar`
+* `rearRightDoorAjar`
 
 #### Updates in MOBILE_API:
 
-##### Update struct `BodyInformation`:
+##### New Struct `DoorsStatus` is needed:
+```xml
+<struct name="DoorStatus" since="X.x">
+	<description>Describes the Status of a parameter of door.</description>
+	<param name="doorLocation" type="Grid" mandatory="true"/>
+	<param name="doorStatus" type="DoorStatusType" mandatory="true"/>
+</struct>
+```
+##### New enum `DoorStatusType` is needed:
+```xml
+<enum name="DoorStatusType" since="X.x">
+	<element name="CLOSED"/>
+	<element name="LOCKED"/>
+	<element name="AJAR"/>
+</enum>
+```
+
+##### New Struct `GatesStatus` is needed:
+```xml
+<struct name="GateStatus" since="X.x">
+	<description>Describes the Status of a parameter of trunk/hood/etc.</description>
+	<param name="gateType" type="GateType" mandatory="true"/>
+	<param name="gateStatus" type="DoorStatusType" mandatory="true"/>
+</struct>
+```
+
+##### New enum `GateType` is needed:
+```xml
+<enum name="GateType" since="X.x">
+	<element name="FRONT"/>
+	<element name="BACK"/>
+	<element name="RIGHT"/>
+	<element name="LEFT"/>
+</enum>
+```
 
-```xml	
+##### Then `BodyInformation` Struct would be updated as:
+```xml
 <struct name="BodyInformation" since="2.0">
 	<param name="parkBrakeActive" type="Boolean" mandatory="true">
-		<description>References signal "PrkBrkActv_B_Actl".</description>
+		<description>true if mechanical park brake is active, false otherwise.</description>
```

Comment:
```suggestion
		<description>True if mechanical park brake is active, false otherwise.</description>
```

Reply:
I actually meant this as response for the param, see this [reference](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/interfaces/MOBILE_API.xml#L4550)

But i'll update this to "If mechanical park brake is active, true. Otherwise false."

Reply:
Done